src/pal/inc/unixasmmacrosx86.inc

```diff
@@ -35,3 +35,40 @@ C_FUNC(\Name\()_End):
         .global C_FUNC(\Name\()_End)
         LEAF_END \Name, \Section
 .endm
+
+.macro push_nonvol_reg Reg
+        push \Reg
+        .cfi_adjust_cfa_offset 4
+        .cfi_rel_offset \Reg, 0
+.endm
+
+.macro pop_nonvol_reg Reg
+        pop \Reg
+        .cfi_restore \Reg
+.endm
+
+.macro PROLOG_BEG
+        .cfi_def_cfa_offset 4
```

Comment:
I would prefer using less .cfi annotations (the more of them there are, the more work the unwinder has). So I would not use push_nonvol_reg here, but rather make the macro as follows:
```asm
.macro PROLOG_BEG
        push ebp
        .cfi_def_cfa_offset 8
        .cfi_offset ebp, -8
        mov ebp, esp
.endm        
```
That's what clang compiler generates